web/src/components/Diagram/components/Timeline/useScrollSpanNodeGroupIntoView.tsx

```diff
@@ -0,0 +1,25 @@
+import {useEffect} from 'react';
+import {useSpan} from '../../../../providers/Span/Span.provider';
+import useControls from '../DAG/hooks/useControls';
+
+interface IProps {
+  isSelected?: boolean;
+  id: string;
+  index: number;
+}
+
+export function useScrollSpanNodeGroupIntoView({index, isSelected, id}: IProps): void {
```

Comment:
Do you mind telling me what's the idea of this hook?

Reply:
it makes sure the timeline bar is visible when the user is stepping through the affected span